thirdparty/nginx/nginx-1.7.1-zimbra/src/core/ngx_zm_lookup.c

```diff
@@ -825,7 +826,12 @@ ngx_zm_lookup_connect (ngx_zm_lookup_ctx_t * ctx)
 #if (NGX_SSL)
 
     if (ctx->handler->ssl && ctx->peer.connection->ssl == NULL) {
-        ngx_zm_lookup_ssl_init_connection(zlcf->ssl, ctx->peer.connection);
+        if(ngx_zm_lookup_ssl_init_connection(zlcf->ssl, ctx->peer.connection) == ZM_LOOKUP_SSL_EVENT_FAILED)
+        {
+            ngx_log_error(NGX_LOG_WARN, ctx->log, 0, "zm lookup: ngx_zm_lookup_connect "
+                    "connect lookup handle error, fail over to the next one");
```

Comment:
@thefoolwhocodes Would it be useful to log the address of the server that failed?

Reply:
This information is available at debug level, but can be added here too.